### PR TITLE
Fixes for `dla-pointer-edges`

### PR DESCRIPTION
**Pull request by AlviseDeFaveri**
This MR contains some commits that are needed by `revng-c`'s `feature/dla-pointer-edges` branch before it can be merged.

### CODE REVIEW

File: include/revng/Model/Binary.h [hunk 1]
```diff
@@ -614,6 +614,7 @@ class model::Binary {
   bool verify(bool Assert) const debug_function;
   bool verify(VerifyHelper &VH) const;
   void dump() const debug_function;
+  void dumpOnFile(const char *FileName) const debug_function;
```
Comment (aleclearmind) [inline include/revng/Model/Binary.h:617]:
```suggestion
  template<typename T>
  void dump(T &Stream) const;
  void dump() const debug_function { dump(dbg); }
```

Comment (AlviseDeFaveri) [reply include/revng/Model/Binary.h:617]: Can I use this inside `gdb` to dump the model at arbitrary points on a file for later inspection?

Comment (aleclearmind) [reply include/revng/Model/Binary.h:617]: Mmmh, we have a `dumpModule` function for `llvm::Module` for `gdb`.
Not sure, I don't want this on every possible part of the model, but maybe we can have it on the root.

Maybe we can have a `std::string toString()` method and then a global function `writeToFile(StringRef Path, StringRef What)`.
```
(gdb) p writeToFile(Model.toString(), "/tmp/asd.yml");
```

What do you think?

Comment (AlviseDeFaveri) [reply include/revng/Model/Binary.h:617]: That would be nice

Comment (AlviseDeFaveri) [reply include/revng/Model/Binary.h:617]: Where is the best place to put `writeToFile()`?

Comment (aleclearmind) [reply include/revng/Model/Binary.h:617]: `include/revng/Support/Debug.h` for now.

Comment (AlviseDeFaveri) [reply include/revng/Model/Binary.h]: I noticed that the order you suggested was different in the gdb example and in the description of the to-be function signature. I went for the former (`writeToFile(StringRef What, StringRef Path)`).

File: include/revng/Support/Debug.h [hunk 2]
```diff
@@ -282,3 +282,6 @@ extern Logger<> NRALog;
 extern Logger<> PassesLog;
 extern Logger<> ReleaseLog;
 extern Logger<> VerifyLog;
+
+void writeToFile(const llvm::StringRef What,
+                 const llvm::StringRef Path) debug_function;
```
Comment (aleclearmind) [inline include/revng/Support/Debug.h:287]: Passing `const` by value does not make a lot of sense.
Also, keep in mind that `StringRef` and `ArrayRef` already represent read-only references (you can't make changes to the pointed data).

Comment (AlviseDeFaveri) [reply include/revng/Support/Debug.h]: oops, fixed removing const.

**Comment by aleclearmind** [comment]
Merged in 2f4b716. Thanks!

**Pull request closed without merging**